lxd/api_vsock.go

```diff
@@ -0,0 +1,229 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/lxc/lxd/lxd/db"
+	"github.com/lxc/lxd/lxd/db/cluster"
+	"github.com/lxc/lxd/lxd/instance"
+	"github.com/lxc/lxd/lxd/project"
+	"github.com/lxc/lxd/lxd/response"
+	"github.com/lxc/lxd/lxd/util"
+	"github.com/lxc/lxd/shared/api"
+	"github.com/lxc/lxd/shared/version"
+)
+
+var vsockAPI10Cmd = APIEndpoint{
+	Get: APIEndpointAction{Handler: vsockAPI10Get, AllowUntrusted: true},
+}
+
+var vsockConfigCmd = APIEndpoint{
+	Path: "config",
+	Get:  APIEndpointAction{Handler: vsockConfigGet, AllowUntrusted: true},
+}
+
+var vsockMetaDataCmd = APIEndpoint{
+	Path: "meta-data",
+	Get:  APIEndpointAction{Handler: vsockMetaDataGet, AllowUntrusted: true},
+}
+
+var vsockDevicesCmd = APIEndpoint{
+	Path: "devices",
+	Get:  APIEndpointAction{Handler: vsockDevicesGet, AllowUntrusted: true},
+}
```

Comment:
Given this is exactly the /dev/lxd API, any hope we can line this stuff up with the actual devlxd API?
It'd be good to not have to double implement everything and for more complex endpoints like /1.0/evens to also use the same code paths.